### ISSUE TITLE
FMU evaluation with dx_refs does not work correctly (incorrect values returned)

### ISSUE THREAD

**Issue by AndreasHofmann217** [ThummeTo/FMICore.jl#72]
Hello Tobi and others,

since FMI standard does not provide method to get individual derivatives, but rather fmiXGetDerivatives the received values from FMICore.jl are incorrect (wrong values are returned).

Find attached a mvp with corresponding FMU and  Dymola model.
Sorry i was unable to get the FMIZoo example working at all :(

Best regards,
Andreas
[mwe_dx_ref.zip](https://github.com/user-attachments/files/16850035/mwe_dx_ref.zip)

**Comment by halentin** [comment]
I just looked into it:
First, i get warnings/errors from the FMU itself when i execute this line: 
```println(fmu(;dx_refs=dx_refs_sub,x=x0)) # should be [-1.5, -2.0] -> actually is [-0.5, - 1.0]```
from your MWE. Those indicate that the results (even though the functions might return some numbers) are probably incorrect.
Those warnings stems from the fact, that `getDerivatives!` is currently not implemented to grab specific derivatives for FMUs in FMIBase.jl. 

I found this TODO:
https://github.com/ThummeTo/FMIBase.jl/blob/07585247d420c54bedfd4a7951428cb96a40b37e/src/common.jl#L213C1-L213C61


Your workaround of getting the whole derivative-vector and selecting the appropriate derivatives after that seems to be the way to go, and i agree that the API is a little bit misleading in that regard.
From my understanding of the standard `fmi3GetContinuousStateDerivatives` only supports getting the whole State-Derivative-Vector, not specific State-Derivatives, so any internal function that gets part of the State-Derivatives would probably need to get the whole Vector anyways. 


From the FMI3 Standard (`fmi3GetContinuousStateDerivatives`): 
"Return argument derivatives contains the derivatives **for each continuous state** with the same convention for the order as defined for fmi3SetContinuousStates."

**Comment by ThummeTo** [comment]
Exactly, FMI-standard itself has no feature for this, and FMI.jl will provide a feature for that (in the near future). It's only a little thing to implement it so we can do that for the next release (writing to a reused buffer and returning a `dx_refs` view to that).
